app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+   License, v. 2.0. If a copy of the MPL was not distributed with this
+   file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.history
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.navigation.Navigation
+import kotlinx.android.synthetic.main.fragment_history.view.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.requireComponents
+import org.mozilla.fenix.mvi.ActionBusFactory
+import org.mozilla.fenix.mvi.getSafeManagedObservable
+import kotlin.coroutines.CoroutineContext
+
+class HistoryFragment : Fragment(), CoroutineScope {
+
+    lateinit var job: Job
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default + job
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_history, container, false)
+        HistoryComponent(view.history_layout, ActionBusFactory.get(this), initialState = HistoryState(emptyList()))
+
+        return view
+    }
+
+    @SuppressLint("CheckResult")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        job = Job()
+
+        setHasOptionsMenu(true)
+        (activity as AppCompatActivity).supportActionBar?.show()
+
+        getSafeManagedObservable<HistoryAction>()
+            .subscribe {
+                if (it is HistoryAction.Select) {
+                    Navigation.findNavController(requireActivity(), R.id.container)
+                        .popBackStack(R.id.browserFragment, false)
+
+                    requireComponents.useCases.sessionUseCases.loadUrl.invoke(it.item.url)
```

Comment:
Do we want loadUrl, which uses the current session? Or do we want tabUseCases.addTab, which creates a new tab? I'm asking because I don't know how the UX is here.

Reply:
Every browser that I've tested uses the existing tab. But this is probably a better question for @shorlander

Reply:
confirmed we should use existing tabs